biz.aQute.tester.junit-platform/src/aQute/tester/bundle/engine/discovery/BundleSelectorResolver.java

```diff
@@ -520,16 +520,14 @@ private boolean elementHasJUnit4Annotations(AnnotatedElement element) {
 		@Override
 		public <T extends DiscoverySelector> List<T> getSelectorsByType(Class<T> selectorType) {
 			info(() -> "Getting selectors from sub-request for: " + selectorType);
-			if (selectorType.equals(ClassSelector.class) || selectorType.equals(MethodSelector.class)) {
-				return selectors.stream()
-					.filter(selectorType::isInstance)
-					.map(selectorType::cast)
-					.collect(toList());
-			}
-			if (selectorType.equals(BundleSelector.class)) {
-				return new ArrayList<>();
-			}
```

Comment:
I'm wondering if it might be a useful optimisation to retain this as a short-circuit just before the main return statement?

Reply:
I don't think this is worth it. It might sometimes be useful with pre 5.5 versions of JUnit 5 but will be counterproductive on later versions.

Reply:
I also found that no one (in our tests) ever calls this method with a `BundleSelector.class` argument anyway.